crates/rattler_cache/src/lib.rs

```diff
@@ -7,9 +8,26 @@ pub mod validation;
 mod consts;
 pub use consts::{PACKAGE_CACHE_DIR, REPODATA_CACHE_DIR};
 
-/// Returns the default cache directory used by rattler.
+/// Determines the default cache directory for rattler.
+/// It first checks the environment variable `RATTLER_CACHE_DIR`.
+/// If not set, it falls back to `XDG_CACHE_HOME/rattler/cache`.
+/// If not set, it falls back to the standard cache directory provided by `dirs::cache_dir()/rattler/cache`.
 pub fn default_cache_dir() -> anyhow::Result<PathBuf> {
-    Ok(dirs::cache_dir()
-        .ok_or_else(|| anyhow::anyhow!("could not determine cache directory for current platform"))?
-        .join("rattler/cache"))
+    Ok(std::env::var("RATTLER_CACHE_DIR")
+        .map(PathBuf::from)
+        .or_else(|_| {
+            std::env::var("XDG_CACHE_HOME")
+                .map(PathBuf::from)
+                .or_else(|_| {
+                    dirs::cache_dir().ok_or_else(|| {
```

Comment:
Im pretty sure that this function already takes the XDG_ variable into account.

Its also linux specific so wouldnt work for windows or macos.